include/nanobind/eigen/sparse.h

```diff
@@ -109,16 +109,26 @@ template <typename T> struct type_caster<T, enable_if_t<is_eigen_sparse_v<T>>> {
             return handle();
         }
 
+        const Index rows = v.rows();
+        const Index cols = v.cols();
         const size_t data_shape[] = { (size_t)v.nonZeros() };
-        const size_t outer_indices_shape[] = { (size_t)((row_major ? v.rows() : v.cols()) + 1) };
-        ScalarTensor data(src.valuePtr(), 1, data_shape);
-        StorageIndexTensor outer_indices(src.outerIndexPtr(), 1, outer_indices_shape);
-        StorageIndexTensor inner_indices(src.innerIndexPtr(), 1, data_shape);
+        const size_t outer_indices_shape[] = { (size_t)((row_major ? rows : cols) + 1) };
+
+        T *src = std::addressof(const_cast<T &>(v));
+        object owner;
+        if (policy == rv_policy::move) {
+            src = new T(std::move(v));
+            owner = capsule(src, [](void *p) noexcept { delete (T *) p; });
+        }
+
+        ScalarTensor data(src->valuePtr(), 1, data_shape, owner);
+        StorageIndexTensor outer_indices(src->outerIndexPtr(), 1, outer_indices_shape, owner);
+        StorageIndexTensor inner_indices(src->innerIndexPtr(), 1, data_shape, owner);
```

Comment:
Do I need one owner per tensor or is this okay? I ran the existing tests with ASan on and there were no failure, but I'm also not sure whether I'm triggering this optimization right now.

Reply:
A single owner shared by all tensors is good. Each tensor will reference the Python capsule, and it can only be deleted when they all expire.